### PR TITLE
mon,osd: dump "compression_algorithms" in "mon metadata"

### PR DESCRIPTION
**Pull request by tchaikov**
Fixes: http://tracker.ceph.com/issues/22420

### CODE REVIEW

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -2067,10 +2068,29 @@ void Monitor::lose_election(epoch_t epoch, set<int> &q, int l,
   finish_election();
 }
 
+namespace {
+std::string collect_compresssion_algorithms()
```
Comment (LenzGr) [inline src/mon/Monitor.cc]: nit: triple "s" in "compresssion"

**Comment by jcsp** [comment]
I think the original idea of reporting content of PluginRegistry was that algorithms declared in Compressor.h are not necessarily present on the system, as they are optional plugins?  However, if we're compiling them all in the same builds in practice then perhaps we could just use the HAVE_LZ4, HAVE_BROTLI cmake variables to populate Compressor::compression_options with what is really built.

Also, isn't the set of algorithms on the OSDs the important thing, not the mons?  We can't assume that presence of something in a mon build means it's present in the OSD build.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-05-04_16:50:59-rados-wip-kefu-testing-2018-05-04-2301-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-05-05_02:02:06-rados-wip-kefu-testing-2018-05-04-2301-distro-basic-mira/

**Comment by chardan** [comment]
Ok... I'm a bit lost in all the dust, but @tchaikov and @jcsp all is well now with respect to required functionality? Is something further needed?

**Comment by jcsp** [comment]
>@jcsp are you suggesting we should send the compression_algorithms in MMgrBeacon?

No, just saying that if the mon is using it's own compiled-in list to restrict selected algorithms, then it's probably also okay to use the mon's metadata field for display in the mgr (i.e. that this patch isn't doing harm).

So I think this patch is OK.

**Comment by tchaikov** [comment]
@jcsp gotcha. may i consider this a LGTM from you?

**Pull request merged**